android/app/src/main/java/com/google/samples/cronet_sample/MainActivity.java

```diff
@@ -123,4 +138,49 @@ public void run() {
             this.cronetLatency.set(averageLatency);
         }
     }
+
+    private synchronized void setCronetEngine() {
+        // Lazily create the Cronet engine.
+        if (cronetEngine == null) {
+            CronetEngine.Builder myBuilder = new CronetEngine.Builder(this);
+            // Enable caching of HTTP data and
+            // other information like QUIC server information, HTTP/2 protocol and QUIC protocol.
+            cronetEngine = myBuilder
+                .enableHttpCache(CronetEngine.Builder.HTTP_CACHE_IN_MEMORY, 100 * 1024)
+                .enableHttp2(true)
+                .enableQuic(true)
+                .build();
+        }
+    }
+
+    /**
+     * Method to start NetLog to log Cronet events.
+     * Find more info about Netlog here:
+     * https://www.chromium.org/developers/design-documents/network-stack/netlog
+     */
+    private void startNetLog() {
+        File outputFile;
+        try {
+            outputFile = File.createTempFile("cronet", "log",
+                this.getExternalFilesDir(null));
+            cronetEngine.startNetLogToFile(outputFile.toString(), false);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Method to properly stop NetLog
+     */
+    private void stopNetLog() {
+        cronetEngine.stopNetLog();
+    }
+    // properly end network logging when app crashes
+    private class ExceptionHandler implements Thread.UncaughtExceptionHandler {
+        public void uncaughtException(Thread thread, Throwable exception) {
+            android.util.Log.e(TAG, exception.getLocalizedMessage());
```

Comment:
Shouldn't this just log the exception in full? As is, the stack trace will be found in a different place.